zerver/tests/test_custom_profile_data.py

```diff
@@ -942,6 +1023,84 @@ def test_default_external_account_type_field(self) -> None:
         )
         self.assert_json_error(result, "Default custom field cannot be updated.")
 
+    def test_update_with_editable_by_user(self) -> None:
+        iago = self.example_user("iago")
+        aaron = self.example_user("aaron")
+        self.login("aaron")
+
+        # Create field with editable_by_user = false
+        dummy_field_data: Dict[str, Any] = {}
+        dummy_field_data["name"] = "Dummy field"
+        dummy_field_data["field_type"] = CustomProfileField.SHORT_TEXT
+        dummy_field_data["editable_by_user"] = "false"
+        result = self.api_post(iago, "/api/v1/realm/profile_fields", info=dummy_field_data)
+        result_json = self.assert_json_success(result)
+        restricted_field_id = result_json["id"]
```

Comment:
`dummy_field_data` feels misleading as a name; maybe `realm_profile_field_data` would be appropriate?